src/mgds/pipelineModules/DownloadHuggingfaceDatasets.py

```
@@ -0,0 +1,54 @@
+import os
+
+from mgds.PipelineModule import PipelineModule
+from mgds.pipelineModuleTypes.RandomAccessPipelineModule import RandomAccessPipelineModule
+import huggingface_hub
+
+
+class DownloadHuggingfaceDatasets(
+    PipelineModule,
+    RandomAccessPipelineModule,
+):
+    def __init__(
+            self,
+            concept_in_name: str, path_in_name: str, enabled_in_name: str,
+            concept_out_name: str,
+    ):
+        super(DownloadHuggingfaceDatasets, self).__init__()
+
+        self.concept_in_name = concept_in_name
+        self.enabled_in_name = enabled_in_name
+        self.path_in_name = path_in_name
+
+        self.concept_out_name = concept_out_name
+
+        self.concepts = []
+
+    def length(self) -> int:
+        return len(self.concepts)
+
+    def get_inputs(self) -> list[str]:
+        return [self.concept_in_name]
+
+    def get_outputs(self) -> list[str]:
+        return [self.concept_out_name]
+
+    def start(self, variation: int):
+        for in_index in range(self._get_previous_length(self.concept_in_name)):
+            concept = self._get_previous_item(variation, self.concept_in_name, in_index)
+            enabled = concept[self.enabled_in_name]
+            if enabled:
+                path = concept[self.path_in_name]
+                is_local = os.path.isdir(path)
+                if not is_local:
```

Can you add a second check to make sure the path is actually a valid huggingface repo id? I think checking if it has exactly one `/` in the name and something before and after should be enough.

---

the best check if a string is potentially a HF repo id appears to be `huggingface_hub.utils.validate_repo_id()`. However, this function is called by `huggingface_hub.download_snapshot()` anyway and a HFValidationError is raised otherwise.

So this is already done by the API, but if your point was that you'd prefer a

`FileNotFoundError: [Errno 2] No such file or directory:`

instead of a
`huggingface_hub.errors.HFValidationError: Repo id must be in the form 'repo_name' or 'namespace/repo_name'`

in case the path is neither a local path, nor a HF repo_id: This is now implemented by ignoring this exception.

---

Good point. I did not see that validation. That should be good enough. I just wanted to avoid sending requests to huggingface just because the path is not a valid local path.